hyperSpec/R/vandermonde.R

```diff
@@ -1,80 +1,82 @@
 ##' Function evaluation on hyperSpec objects
 ##'
-##' vandermonde generates van der Monde matrices, the hyperSpec method generates a hyperSpec object
-##' containing the van der Monde matrix of the wavelengths of a hyperSpec object.
+##' `vandermonde()` generates Vandermonde matrices, the hyperSpec method
+##' generates a hyperSpec object containing the van der Monde matrix of the
+##' wavelengths of a hyperSpec object.
 ##'
-##' It is often numerically preferrable to map \code{wl (x)} to [0, 1], see the example.
+##' It is often numerically preferrable to map `wl(x)` to [0, 1], see the
+##' example.
 ##'
 ##' @param x object to evaluate the polynomial on
 ##' @param order of the polynomial
+##' @md
 ##' @rdname vanderMonde
-##' @return van der Monde matrix
+##' @return Vandermonde matrix
 ##' @author C. Beleites
 ##' @export
 ##' @include unittest.R
-vanderMonde <- function (x, order, ...){
-  if (nargs () > 2)
-    stop ('Unknown arguments: ', names (c (...)))
+vanderMonde <- function(x, order, ...) {
+  if (nargs() > 2)
+    stop('Unknown arguments: ', names(c (...)))
 
-  outer (x, 0 : order, `^`)
+  outer(x, 0:order, `^`)
 }
 
 ##' @noRd
 setGeneric ("vanderMonde")
 
 ##' @param normalize.wl function to transorm the wavelengths before evaluating the polynomial (or
-##' other function). \code{\link[hyperSpec]{normalize01}} maps the wavelength range to the interval
-##' [0, 1]. Use \code{\link[base]{I}} to turn off.
-##' @param ... hyperSpec method: further arguments to \code{\link{decomposition}}
-##' @return hyperSpec method: hyperSpec object containing van der Monde matrix as spectra and an additional column ".vdm.order" giving the order of each spectrum (term).
+##' other function). `hyperSpec::normalize01()` maps the wavelength range to the interval
+##' [0, 1]. Use `base::I()` to turn off.
```

Comment:
I suggest using `[base::I()]` (square brackets instead of backticks) as this syntax will automatically create a link to the documentation of function `I()`. 
The same applies to `[hyperSpec::decomposition()]` and  the other functions.

Reply:
Especially as the previous Rd code was `\code{\link[base]{I}} ` (i.e., defined a link)

Reply:
@GegznaV ...automatically creates a link to the documentation... Where?  In `pkgdown`? Not sure what kind of text processing you are thinking about.

Reply:
In R documentation as well as in `pkgdown` website.

Here is more on links in Markdown-favored R documentation: 
https://roxygen2.r-lib.org/articles/markdown.html#links

Reply:
This markdown syntax and `roxygen` comments (`#'`, or `##'` in **hyperSpec**'s case) indicate that **roxygen2** is used to create Rd files automatically. And syntax `[pkg::func()]` in the *roxygen* comments is converted into `\code{\link[pkg:func]{pkg::func()}}` in appropriate Rd file. Finally, `\code{\link[pkg:func]{pkg::func()}}` is converted into a link when  documentation is built from Rd file.

Reply:
Fantastic I did not know this trick.  Thank you!  Not only saves typing but when you do it all manually it's easy to miss a closing brace or square bracket.  Fabulous!

Reply:
If you use RStudio, you may notice, that text inside square brackets inside _roxygen_ comments is highlighted differently than the surrounding text.
Of course, these corrections in the documentation are not crucial. I just draw your attention that current changes in syntax (using just backticks) will not result in the links, which were present in the documentation when LaTeX based Rd syntax was used.

Reply:
There is a package, that converts from Rd to markdown syntax in roxygen comments: [**roxygen2md**](https://github.com/r-lib/roxygen2md).